src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -36,37 +36,50 @@
 from nomad_simulations.schema_packages.utils import is_not_representative
 
 
-class ModelMethod(ArchiveSection):
+class BaseModelMethod(ArchiveSection):
     """
-    Model method input parameters and numerical settings used to simulate materials properties.
-
-    # ! add more description once section is finished
+    A base section used to define the abstract class of a Hamiltonian section. This section is an
+    abstraction of the `ModelMethod` section, which contains the settings and parameters used in
+    the mathematical model solved in a simulation. This abstraction is needed in order to allow
+    `ModelMethod` to be divided into specific `terms`, so that the total Hamiltonian is specified in
+    `ModelMethod`, while its contributions are defined in `ModelMethod.terms`.
+
+    Example: a custom model Hamiltonian containing two terms:
+        $H = H_{V_{1}(r)}+H_{V_{2}(r)}$
+    where $H_{V_{1}(r)}$ and $H_{V_{2}(r)}$ are the contributions of two different potentials written in
+    real space coordinates $r$. These potentials could be defined in terms of a combination of parameters
+    $(a_{1}, b_{1}, c_{1}...)$ for $V_{1}(r)$ and $(a_{2}, b_{2}, c_{2}...)$ for $V_{2}(r)$. If we name the
+    total Hamiltonian as `'FF1'`:
+        `ModelMethod.name = 'FF1'`
+        `ModelMethod.contributions = [BaseModelMethod(name='V1', parameters=[a1, b1, c1]), BaseModelMethod(name='V2', parameters=[a2, b2, c2])]`
```

Comment:
from this example it is unclear to me if parameters will be defined at the level of BaseModelMethod() or only within subclasses?

I had done something like this before for Force Fields:
```python
class Potential(ArchiveSection):
    """
    Section containing information about an interaction potential.
    """

    type = Quantity(
        type=str,
        shape=[],
        description="""
        Specifies the type (i.e., functional form) of the interaction potential.
        """,
    )

    parameters = Quantity(
        type=dict[str, np.float64], 
        shape=[],
        description="""
        Dictionary of label and parameters of the interaction potential.
        """,
    )

    def normalize(self, archive, logger) -> None:
        super().normalize(archive, logger)

class BondPotential(Potential):
    """
    Section containing information about bond potentials.

    Suggested types are: harmonic, cubic, Morse, fene, tabulated
    """

    equilibrium_value = Quantity(
        type=np.float64,
        unit='m',
        shape=[],
        description="""
        Specifies the equilibrium bond distance.
        """,
    )

    force_constant = Quantity(
        type=np.float64,
        shape=[],
        unit='J / m **2',
        description="""
        Specifies the force constant of the bond potential.
        """,
    )

    def normalize(self, archive, logger) -> None:
        super().normalize(archive, logger)
```

where I define some default structure for parameters, but then define specific, named parameters for particular types of potentials. But in this case `parameters` becomes obsolete, so now I don't think it really makes sense to define it at this abstract level (or even at the level of `Potential()`). If you do define it at this abstract level I think you need a structure with labels/descriptors and values, right?

Reply:
I think indeed the parameters could be first defined in the specific classes that inherit from `BaseModelMethod`, and then we can check later on if there are commonalities or a way of standardizing these into the abstract class.

So in your example, you just need to define `BondPotential(BaseModelMethod)` (and the parameters as a `type=np.float64`, I guess and if you want, or only the `equilibrium_value` and `force_constant`) and `ForceField(ModelMethod)`.

Does this sound good?